lib/pbench/test/unit/server/test_datasets_inventory.py

```diff
@@ -43,11 +51,15 @@ def query_api(
     def mock_find_dataset(self, dataset):
         class Tarball(object):
             unpacked = Path("/dataset1/")
+            tarball_path = Path("/dataset1/")
 
         # Validate the resource_id
         Dataset.query(resource_id=dataset)
         return Tarball
 
+    def mock_send_file(*args, **kwargs) -> Dict[str, str]:
+        return {"status": "OK"}
```

Comment:
Define a class variable, like

```
class TestDatasetsAccess:
    send_file = None
```

Then, here,

```
    def mock_send_file(*args, **kwargs) -> Dict[str, str]:
        TestDatasetsAccess.send_file = args[0]
        return {"status": "OK"}
```

And then in `test_get_result_tarball` you can `assert` that `__class__.send_file == "/dataset1/"`. (This is even a slightly better test if your `unpacked` and `tarball_path` mock values are different so you can tell them apart. 😁 )

You would use it similarly in the other test cases to check the file path that was "downloaded" relative to your unpack directory.

Reply:
My concern here is that I'm not sure what the lifetime of the class variable is.  That is, it may survive from one test to another in the same class, in which case, each test should initialize the variable so that it can determine not only that it ended up with the right value but also that it was actually _set_ to that value....

Given that the mock function is so simple, I would recommend defining it inside the function where it is used.  Yeah, we end up repeating it everywhere where it is used, but the local definition means that it can reference local variables in the outer function, and those have well-understood scopes and lifetimes.

So, it would look something like this:
```
    def test_dataset_in_given_path(self, query_get_as, monkeypatch):
        file_sent = None

        def mock_send_file(path, *args, **kwargs):
            file_sent = path
            return {"status": "OK"}

        monkeypatch.setattr(FileTree, "find_dataset", self.mock_find_dataset)
        monkeypatch.setattr(Path, "is_file", lambda self: True)
        monkeypatch.setattr(werkzeug.utils, "send_file", mock_send_file)

        response = query_get_as("fio_2", "1-default/default.csv", HTTPStatus.OK)

        assert response.status_code == HTTPStatus.OK
        assert file_sent == "1-default/default.csv"
```

Reply:
It's not that hard to reset the class variable (which I believe does survive across test cases given my previous experience, although I deliberately decided not to sketch out too much detail 😬 ), and I generally dislike this sort of repetition (even when it's simple) ... but it should work.

(Note: see the `TestUpload` class in `test_requests.py` for an example.)